crates/sui-core/src/authority.rs

```diff
@@ -555,6 +556,9 @@ pub struct AuthorityState {
 
     pub metrics: Arc<AuthorityMetrics>,
 
+    /// In-memory cache of the content from the reconfig_state db table.
+    reconfig_state_mem: tokio::sync::Mutex<ReconfigState>,
```

Comment:
it doesn't look like this needs to be a tokio mutex, since you never hold it across an await.

i also wonder if it is worth making this an RwLock - it looks like there could be a fair amount of contention caused by AuthorityServer.

Reply:
I suspect it might be useful to have a ReadWriteLock here actually (and I do think that some future use cases are likely to need it to be tokio lock, but I am also ok with changing it when needed)